### PR TITLE
Update API section

### PR DESCRIPTION
**Pull request by quasiben**
This PR updates the API section of the docs

### CODE REVIEW

File: docs/source/api.rst [hunk 1]
```diff
@@ -1,9 +1,71 @@
...
    ucp
    ucp.create_listener
-   ucp.create_endpoint
\ No newline at end of file
+   ucp.create_endpoint
+   ucp.init
+   ucp.get_address
+   ucp.get_config
+   ucp.get_ucp_worker
+   ucp.get_ucx_version
```
Comment (pentschev) [inline docs/source/api.rst]:
```suggestion
   ucp.get_ucx_version
   ucp.init
   ucp.progress
```

File: docs/source/api.rst [hunk 2]
```diff
@@ -1,9 +1,71 @@
...
+
+.. autosummary::
+   Endpoint
+   Endpoint.abort
+   Endpoint.cuda_support
+   Endpoint.get_ucp_endpoint
+   Endpoint.get_ucp_endpoint
+   Endpoint.close
+   Endpoint.closed
+   Endpoint.close_after_n_recv
```
Comment (pentschev) [inline docs/source/api.rst]:
```suggestion
   Endpoint.close_after_n_recv
   Endpoint.cuda_support
   Endpoint.get_ucp_endpoint
   Endpoint.get_ucp_worker
```

File: docs/source/api.rst [hunk 3]
```diff
@@ -1,9 +1,71 @@
...
+   ucp.get_ucp_worker
+   ucp.get_ucx_version
+   ucp.reset
+
+**Endpoint**
+
+.. autosummary::
+   Endpoint
+   Endpoint.abort
+   Endpoint.cuda_support
```
Comment (pentschev) [inline docs/source/api.rst]:
```suggestion
```

File: docs/source/api.rst [hunk 4]
```diff
@@ -1,9 +1,71 @@
...
+   Endpoint.get_ucp_endpoint
```
Comment (pentschev) [inline docs/source/api.rst]:
```suggestion
```

File: docs/source/api.rst [hunk 5]
```diff
@@ -1,9 +1,71 @@
...
+   Endpoint.get_ucp_endpoint
```
Comment (pentschev) [inline docs/source/api.rst]:
```suggestion
```

File: docs/source/api.rst [hunk 6]
```diff
@@ -1,9 +1,71 @@
...
+**ucp**
 
 .. autosummary::
+
    ucp
    ucp.create_listener
-   ucp.create_endpoint
\ No newline at end of file
+   ucp.create_endpoint
+   ucp.init
```
Comment (pentschev) [inline docs/source/api.rst]:
```suggestion
```

File: docs/source/api.rst [hunk 7]
```diff
@@ -1,9 +1,71 @@
...
+
+.. currentmodule:: ucp
+
+.. autofunction:: create_listener
+.. autofunction:: create_endpoint
+.. autofunction:: get_address
+.. autofunction:: get_config
+.. autofunction:: get_ucp_worker
+.. autofunction:: get_ucx_version
+.. autofunction:: init
```
Comment (pentschev) [inline docs/source/api.rst]:
```suggestion
.. autofunction:: init
.. autofunction:: progress
```

File: docs/source/api.rst [hunk 8]
```diff
@@ -1,9 +1,71 @@
 API
 ===
 
-.. currentmodule:: ucx_py
+.. currentmodule:: ucp
+
+**ucp**
 
 .. autosummary::
+
```
Comment (jakirkham) [inline docs/source/api.rst]: Should we have a blank line like this for other directives or should we drop this one? Only curious if this impacts whether the docs build correctly or not.

Comment (quasiben) [reply docs/source/api.rst]: it doesn't matter but I'll fix to be consistent.  Thanks

**Comment by quasiben** [comment]
Thanks @pentschev @jakirkham comments have been merged in

File: docs/source/api.rst (hunk 8 above)
Comment (jakirkham) [reply docs/source/api.rst]: Ok sorry for the nit then.

**Comment by jakirkham** [comment]
Thanks @quasiben. Looks ready to go on my end! 😄

**Comment by pentschev** [comment]
LGTM too! Thanks @quasiben

**Pull request merged**